### DISCUSSION TITLE
After upgrading Flutter, the integration tests on the iOS emulator crash

### DISCUSSION THREAD

**Discussion by tformatix** [codemagic-ci-cd/codemagic-docs#2649] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. Upgrade from Flutter 3.19.1 to Flutter 3.22.3 (`flutter upgrade`)
2. Upgrade packages with `flutter pub upgrade`
3. Change from `Theme.of(context).colorScheme.background` to `Theme.of(context).colorScheme.surface` as `background` is deprecated
4. Fixing other integration tests that did not work with the new version (too little scrolling and missing/wrong mocks) 

### Expected results

All integration tests should be run as before.

### Actual results

Apparently the tests crash at a certain point. The tests crash at different points (somehow different, sometimes they are the same). Rarely do all tests run successfully.

**This is the error message:**
```
Unhandled exception:
DriverError: Failed to fulfill RequestData due to remote error
Original error: ext.flutter.driver: (112) Service has disappeared
Original stack trace:
#0      new _OutstandingRequest (package:vm_service/src/vm_service.dart:265:34)
#1      VmService._call.<anonymous closure> (package:vm_service/src/vm_service.dart:1772:25)
#2      VmService._call (package:vm_service/src/vm_service.dart:1784:8)
#3      VmService.callServiceExtension (package:vm_service/src/vm_service.dart:1733:14)
#4      VMServiceFlutterDriver.sendCommand (package:flutter_driver/src/driver/vmservice_driver.dart:322:66)
#5      FlutterDriver.requestData (package:flutter_driver/src/driver/driver.dart:540:45)
#6      integrationDriver (package:integration_test/integration_test_driver_extended.dart:96:38)
<asynchronous suspension>
#7      main (file:///Users/builder/clone/flutter/App/test_driver/integration_test.dart:5:3)
<asynchronous suspension>


#0      VMServiceFlutterDriver.sendCommand (package:flutter_driver/src/driver/vmservice_driver.dart:334:7)
<asynchronous suspension>
#1      FlutterDriver.requestData (package:flutter_driver/src/driver/driver.dart:540:39)
<asynchronous suspension>
#2      integrationDriver (package:integration_test/integration_test_driver_extended.dart:96:25)
<asynchronous suspension>
#3      main (file:///Users/builder/clone/flutter/App/test_driver/integration_test.dart:5:3)
<asynchronous suspension>
Unable to terminate com.example.app on D3F41A76-A956-4F9B-8874-99974B22C923:
ProcessException: Process exited abnormally with exit code 3:
An error was encountered processing the command (domain=NSPOSIXErrorDomain, code=3):
Simulator device failed to terminate com.example.app.
found nothing to terminate
Underlying error (domain=NSPOSIXErrorDomain, code=3):
	The request to terminate "com.example.app" failed. found nothing to terminate
	found nothing to terminate
  Command: /usr/bin/arch -arm64e xcrun simctl terminate D3F41A76-A956-4F9B-8874-99974B22C923 com.example.app


QA failed :|
Flutter Drive run failed.

== QA failed, ending build ==


Build failed :|
Test run failed: Flutter Drive run failed.
```

### Build id (optional)

66a273c6a595665195379974

**Comment by dtrdic** [comment, marked as answer]
hey @tformatix try localy to remove `pubspec.lock` and `podfile.lock` files then run `flutter pub get` and `pod install`. New .lock files will be generated, then try your tests again. If everything is ok push the changes and re-run the build

**Reply by tformatix** [reply]
Hey @dtrdic, thank you for your fast replay. Seems to be working now. Sorry to have wasted your time - this is an issue we could have considered as well.

**Reply by dtrdic** [reply]
no problem, i'm glad it worked:)
